File: examples/modify_chair.py
"""Edit the previously saved chair revision in place: recolor the cushion and
add a stitched tuft mark, without resending any row the chair.py script
already authored. Demonstrates PixelArt.load(bpy.context.scene) for reading
back an existing scene's pixel_art definition inside execute_blender_python."""

import bpy

from pixel_art_mcp.pixel_art import Canvas, PixelArt

art = PixelArt.load(bpy.context.scene)
art.palette["C"] = "#aa466b"  # recolor the cushion pad from tan to a dusty rose
for layer in art.layers:
    if layer["name"] != "body":
        continue
    for pose in layer["poses"]:
        canvas = Canvas.from_rows(pose["rows"])
        if pose["angle"] in (0, 180):
            canvas.rect(6, 12, 4, 1, "D").rect(7, 13, 2, 3, "D")  # a small tufted stitch
        else:
            canvas.rect(5, 12, 2, 3, "D")
        pose["rows"] = canvas.rows
art.save(bpy.context.scene)

File: examples/modify_chair.ts
import { Canvas, PixelArt } from "@pixel-art-mcp/pixel-core";

/**
 * Edit the previous saved chair revision, retaining named views and layers.
 *
 * Ported from `modify_chair.py`, which reads the previous revision via `PixelArt.load(scene)`
 * before mutating it -- the TS port uses `PixelArt.load`/`PixelArt.save` from
 * `@pixel-art-mcp/pixel-core` the same way. `pose.rows`/`layer.name` are the same plain
 * snake_case fields `pixel-core`'s `toDict()`/`fromDict()` round-trip through (see
 * `packages/pixel-core/src/types.ts`), so mutating them in place and calling `art.save(scene)`
 * is the direct equivalent of the Python source's in-place dict mutation.
 */
export default function main(scene: Scene): void {
  const art = PixelArt.load(scene);
  art.palette["C"] = "#aa466b";
  art.palette["T"] = "#e69eb0";
  for (const layer of art.layers) {
    if (layer.name !== "body") continue;
    for (const pose of layer.poses) {
      const canvas = Canvas.fromRows(pose.rows);
      if (pose.angle === 0 || pose.angle === 180) {
        canvas.rect(3, 8, 10, 2, "D").rect(4, 8, 8, 1, "H");
        canvas.rect(4, 9, 8, 2, "C").rect(5, 9, 6, 1, "T");
      } else {
        const x = pose.angle === 90 ? 2 : 11;
        canvas.rect(x, 8, 3, 3, "D").rect(x + 1, 9, 1, 2, "H");
      }
      pose.rows = canvas.rows;
    }
  }
  art.save(scene);
}
